### ISSUE TITLE
AutoExecute without database cache produces high mysql load

### ISSUE THREAD

**Issue by Nobodys** [ADOdb/ADOdb#434]
I'm working on a project where the cronjob is running without database cache as it runs parallel with other cronjobs.

For tables with some data (30k entries) the AutoExecute function creates a lot of mysql load as it always does a "SELECT * FROM $table LIMIT 1" query on every insert.

Although the query is limited to 1 it still creates load on mysql. Would it not be faster/better to use database "information_schema" is available?

**Comment by mnewnham** [comment]
First, I'm going to quote the warning note at the top of manual page for [AutoExecute](http://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:autoexecute)

> autoExecute is designed to provide a robust simple interface to record updating and insertion. As such, it contains multiple layers of table, column and data validation to ensure data integrity prior to record insertion or update. Consequently, there may be some performance degradation when used for large numbers of record updates. Consider using execute() for that purpose instead. 

Second, I'm going to say that if you want to do bulk inserting, autoexecute is literally the worst thing to use for such a process. Even [getInsertSql](http://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:getinsertsql), for which autoExecute is just a wrapper, would likely run 2-3 times faster and is almost as easy to use.

Lastly, you should look at [execute](http://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:execute), which is the recommended method for this type of process, and, depending on your database, might give you access to  [Bulk Binding](http://adodb.org/dokuwiki/doku.php?id=v5:userguide:learn_bind:bulkbinding), which will run (figuratively) a million times faster.

All that being said, caching dictionary information is always a good idea, and if you wanted a project that interests you to contribute, we are more than happy to accept such a thing. Just remember that there are reasons for not caching and so a feature would need to be switchable.

In summary, the project team would likely not be interested in spending time doing performance tweaks on autoExecute, given its designated usage, but we would be more than happy for you to contribute such an item.

**Comment by mnewnham** [comment]
As I have not had a response from you in 10 days, I am closing this issue. If you need more help on the subject, feel free to reopen it. If you can contribute some time and code on the issue, that would be great
